src/openrct2/ride/Vehicle.h

```diff
@@ -111,6 +111,41 @@ enum VEHICLE_TYPE : uint8_t
     VEHICLE_TYPE_TAIL = 1,
 };
 
+enum VEHICLE_STATUS
```

Comment:
I think it would be better to use enum class with uint8_t as type.

Reply:
I was doing some reading about enum classes earlier today. I wasn't going to suggest it because no other enums in vehicle.h use them, and I'm still very much learning my around c++. However, It would allow s4import and s6import to make safer range checks.

Reply:
Most of the code is from the days where everything was compiled as C, now that we compile everything as C++ we should use enum class wherever appropriate. It should look like following:
```
enum class VehicleStatus : uint8_t
{
     MovingToEndOfStation = 0,
     WaitingForPassengers,
     ...
};
```

Reply:
I can do that. I'll also figure out how to range check appropriately, so adding or removing enums later on doesn't break anything.

Reply:
I was going to leave changing it to an enum class for a second pr. I think it requires quite a good few changes and I'd prefer to have this pr just for set state.

Reply:
Oh I kinda figured since the lines are already affected it might be a good time but another PR works for me aswell

Reply:
Your forgetting all of the branching code that checks the status. There will be a lot of it scattered throughout the code base.

Reply:
@duncanspumpkin Was `Your forgetting all of the branching code` directed at @ZehMatt or me? I did find one location where we were still checking the status by integer value, but that's been fixed. (Will be pushed shortly).

Reply:
It was directed at zehmatt.

Reply:
Ok. I'll push the recent change. If the tests fail again, what do we do? I was unable to decipher anything meaningful out of the logs. I've never used Travis-ci before, not really sure what to look for.

Reply:
Just look at the log for errors/warnings https://travis-ci.org/OpenRCT2/OpenRCT2/jobs/540830525 and address them.